### PR TITLE
Add touch and hover event support to BpkBarchart

### PR DESCRIPTION
**Pull request by tomcorke**
+ [x] For any new components I've made sure that the style can be extended by the consumer using a `className` override.
+ [x] For any new files I've included the [Apache 2.0 License header](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-tokens/formatters/license-header.js#L2-L16) at the top of the file.
+ [x] I've updated `changelog.md` for any changes that need to be highlighted in the changelog.
+ [x] If I've updated `npm-shrinkwrap.json` those changes are intentional.

### CODE REVIEW

File: packages/bpk-component-barchart/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bpk-component-barchart",
-  "version": "2.0.12",
+  "version": "2.1.0",
```
Comment (k0nserv) [inline packages/bpk-component-barchart/package.json]: Please don't update these version numbers. Lerna takes care of that for us

File: changelog.md [hunk 2]
```diff
@@ -13,6 +13,8 @@
 ## 2017-09-20 - New React Native Button component
 
 **Added:**
+- bpk-component-barchart: 2.0.10 => 2.1.0
```
Comment (k0nserv) [inline changelog.md]: Don't reference the new version number here, we do this as part of the release

**Comment by tomcorke** [comment]
Thanks @k0nserv, I've removed those changes.

File: packages/bpk-component-barchart/package.json.orig [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "bpk-component-barchart",
+<<<<<<< HEAD
```
Comment (georgegillams) [inline packages/bpk-component-barchart/package.json.orig]: Unresolved conflict

**Comment by georgegillams** [comment]
I feel like we should include examples of the different interaction-types in storybook. We can do that  ourselves as a separate PR, though 😸

**Pull request merged**